ci/lint.sh

```diff
@@ -1,196 +1,123 @@
-        if [ $? -ne "0" ]; then
-            RET=1
-        fi
-    done
-    echo "linting -> pandas/_libs/tslibs/src/datetime"
-    cpplint --quiet --extensions=c,h --headers=h --filter=-readability/casting,-runtime/int,-build/include_subdir --recursive pandas/_libs/tslibs/src/datetime
-    if [ $? -ne "0" ]; then
-        RET=1
-    fi
-    echo "Linting *.c and *.h DONE"
-
-    echo "Check for invalid testing"
-
-    # Check for the following code in testing:
-    #
-    # np.testing
-    # np.array_equal
-    grep -r -E --include '*.py' --exclude testing.py '(numpy|np)(\.testing|\.array_equal)' pandas/tests/
-
-    if [ $? = "0" ]; then
-        RET=1
-    fi
-
-    # Check for pytest.warns
-    grep -r -E --include '*.py' 'pytest\.warns' pandas/tests/
-
-    if [ $? = "0" ]; then
-        RET=1
-    fi
-
-    # Check for the following code in the extension array base tests
-    # tm.assert_frame_equal
-    # tm.assert_series_equal
-    grep -r -E --include '*.py' --exclude base.py 'tm.assert_(series|frame)_equal' pandas/tests/extension/base
-
-    if [ $? = "0" ]; then
-        RET=1
-    fi
-
-    echo "Check for invalid testing DONE"
-
-    # Check for imports from pandas.core.common instead
-    # of `import pandas.core.common as com`
-    echo "Check for non-standard imports"
-    grep -R --include="*.py*" -E "from pandas.core.common import " pandas
-    if [ $? = "0" ]; then
-        RET=1
-    fi
-    echo "Check for non-standard imports DONE"
-
-    echo "Check for incorrect sphinx directives"
-    SPHINX_DIRECTIVES=$(echo \
-       "autosummary|contents|currentmodule|deprecated|function|image|"\
-       "important|include|ipython|literalinclude|math|module|note|raw|"\
-       "seealso|toctree|versionadded|versionchanged|warning" | tr -d "[:space:]")
-    for path in './pandas' './doc/source'
-    do
-        grep -R --include="*.py" --include="*.pyx" --include="*.rst" -E "\.\. ($SPHINX_DIRECTIVES):[^:]" $path
-        if [ $? = "0" ]; then
-            RET=1
-        fi
-    done
-    echo "Check for incorrect sphinx directives DONE"
-
-    echo "Check for deprecated messages without sphinx directive"
-    grep -R --include="*.py" --include="*.pyx" -E "(DEPRECATED|DEPRECATE|Deprecated)(:|,|\.)" pandas
-
-    if [ $? = "0" ]; then
-        RET=1
-    fi
-    echo "Check for deprecated messages without sphinx directive DONE"
-
-    echo "Check for old-style classes"
-    grep -R --include="*.py" -E "class\s\S*[^)]:" pandas scripts
-
-    if [ $? = "0" ]; then
-        RET=1
-    fi
-    echo "Check for old-style classes DONE"
-
-    echo "Check for backticks incorrectly rendering because of missing spaces"
-    grep -R --include="*.rst" -E "[a-zA-Z0-9]\`\`?[a-zA-Z0-9]" doc/source/
-
-    if [ $? = "0" ]; then
-        RET=1
-    fi
-    echo "Check for backticks incorrectly rendering because of missing spaces DONE"
-
-else
-    echo "NOT Linting"
-fi
+
+### LINTING ###
```

Comment:
so I would rather have these as 3 scripts, then a top-level script to call each one. That way we can call these individually (e.g. in the Makefile)

Reply:
You mean one script for linting (flake8+cpplint), one for the checks (grep) and one for the doctests, right? I think that makes sense, but considering we may want to move this to azure, I see different options:
- Having 3 scripts as you say
- Having one, but with a parameter (e.g. `./ci/lint.sh [lint|checks|doctests]`). I think this makes things a bit simpler and more compact (as the 3 scripts will be very similar)
- Leave it as it is, and move in a separate PR the content of this file to `azure-pipelines.yml`

You'll know better than me how often these checks are run locally or out of the CI, but as now flake8 is mainly `flake8 .` and the doctests will soon be `flake 8 . --doctests` I think it can make sense to just have them in azure. May be a script just for the `grep` checks if you think those are being run locally?